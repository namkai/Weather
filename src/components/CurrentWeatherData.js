import PropTypes from 'prop-types';
import React from 'react';
import '../style/CurrentWeatherData.css';

const CurrentWeatherData = ({ main, wind, sys }) => (
	<div className="col-md-6">
		<div className="jumbotron jumbotron-fluid" style={{ padding: '1.2rem 1.5rem', textAlign: 'center' }}>
			<div className="container">
				<h1 className="display-5">{Math.floor(main.temp * 9 / 5 - 459.67)} &deg;F</h1>
			</div>
		</div>
		<table className="table table-inverse">
			<tbody>
			<tr>
				<th scope="row">Humidity</th>
				<td>{main.humidity}%</td>
			</tr>
			<tr>
				<th scope="row">Pressure</th>
				<td>{main.pressure}</td>
			</tr>
			<tr>
				<th scope="row">Wind Speed</th>
				<td>{wind.speed} mph</td>
			</tr>
			<tr>
				<th scope="row">Sunrise</th>
				<td>{sys.sunrise}</td>
			</tr>
			<tr>
				<th scope="row">Sunset</th>
				<td>{sys.sunset}</td>
			</tr>
			</tbody>
		</table>
	</div>
);

CurrentWeatherData.defaultProps = {
	main: {
		temp: PropTypes.number,
		humidity: PropTypes.string,
		pressure: PropTypes.string,
	},
	name: PropTypes.string,
	wind: PropTypes.object,
	weather: PropTypes.array,
	sys: {
		sunrise: PropTypes.number,
		sunset: PropTypes.number,
	},
};

CurrentWeatherData.propTypes = {
	main: PropTypes.object,
	name: PropTypes.string,
	weather: PropTypes.array,
	wind: PropTypes.object,
	sys: PropTypes.object,
};

export default CurrentWeatherData;
