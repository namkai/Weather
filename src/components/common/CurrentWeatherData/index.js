import CurrentWeatherData from './view';

export default CurrentWeatherData;

