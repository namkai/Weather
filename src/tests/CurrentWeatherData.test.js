import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import CurrentWeatherData from '../components/CurrentWeatherData';

it('renders correctly', () => {
	const currentweatherdata = shallow(<CurrentWeatherData />);
	expect(toJson(currentweatherdata)).toMatchSnapshot();
});