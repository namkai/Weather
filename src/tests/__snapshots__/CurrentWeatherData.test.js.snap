// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="col-md-6"
>
  <div
    className="jumbotron jumbotron-fluid"
    style={
      Object {
        "padding": "1.2rem 1.5rem",
        "textAlign": "center",
      }
    }
  >
    <div
      className="container"
    >
      <h1
        className="display-5"
      >
        NaN
         Â°F
      </h1>
    </div>
  </div>
  <table
    className="table table-inverse"
  >
    <tbody
      style={
        Object {
          "textAlign": "right",
        }
      }
    >
      <tr>
        <th
          scope="row"
        >
          <i
            aria-hidden="true"
            className="fa fa-cloud"
          />
            Humidity
        </th>
        <td>
          %
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          <span
            className="icon icon-thermometer"
          />
            Pressure
        </th>
        <td>
           mb
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          <span
            className="icon icon-air"
          />
            Wind Speed
        </th>
        <td>
           mph
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          <span
            className="icon icon-light-up"
          />
            Sunrise
        </th>
        <td>
          NaN:NaN AM
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          <span
            className="icon icon-light-down"
          />
            Sunset
        </th>
        <td>
          NaN:NaN PM
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
